packages/next/client/image.tsx

```diff
@@ -325,14 +326,17 @@ function handleLoading(
     }
   }
   if (imgRef.current) {
-    if (imgRef.current.complete) {
```

Comment:
I think we might want to still rely on `imgRef.current.complete` for the initial render and ensure one `onload`/`onerror` listener is added to the `img` element for use after that. Then we can track which items are already loaded in the existing module scoped `Set` (loadedImageURLs). This should handle the test case that failed [here](https://github.com/vercel/next.js/runs/5825372547?check_suite_focus=true#step:9:2106)